examples/c/publish_subscribe_cross_language/README.md

```diff
@@ -0,0 +1,110 @@
+# Cross-language Publish-Subscribe
+
+Before proceeding, all dependencies need to be installed. You can find
+instructions in the [C Examples Readme](../README.md).
+
+## Running The Example
+
+This example illustrates a robust publisher-subscriber communication pattern
+between a C publisher and a Rust subcriber application. The C publisher
+application sends a message every second, each containing `TransmissionData` and
+the `CustomHeader`. On the receiving end, the Rust subscriber application prints
+the received payload and the user header to the console whenever new data
+arrives.
+
+First you have to build the C examples:
+
+```sh
+cmake -S . -B target/ffi/build -DBUILD_EXAMPLES=ON
+cmake --build target/ffi/build
+```
+
+To observe this dynamic communication in action, open two separate terminals and
+execute the following commands:
+
+### Terminal 1
+
+Run the Rust subscriber application from the
+[Rust cross-language publish-subscribe example](../../rust/publish_subscribe_cross_language):
+
+```sh
+cargo run --example publish_subscribe_cross_language_subscriber
+```
+
+### Terminal 2
+
+Run the C publisher application:
+
+```sh
+./target/ffi/build/examples/c/publish_subscribe_cross_language/example_c_cross_language_publisher
+```
+
+Feel free to also run the C subscriber and the Rust publisher applications
+simultaneously to explore how iceoryx2 handles cross-language
+publisher-subscriber communication efficiently.
+
+### Terminal 3
+
+Run the C subscriber application:
+
+```sh
+./target/ffi/build/examples/c/publish_subscribe_cross_language/example_c_cross_language_subscriber
+```
+
+### Terminal 4
+
+Run the Rust publisher application from the
+[Rust cross-language publish-subscribe example](../../rust/publish_subscribe_cross_language):
+
+```sh
+cargo run --example publish_subscribe_cross_language_publisher
+```
+
+You can also communicate with the C++ publisher and subscriber applications from
+the
+[C++ cross-language publish-subscribe example](../../cxx/publish_subscribe_cross_language).
+
+You may hit the maximum supported number of ports when too many publisher or
+subscriber processes are running. Check the [iceoryx2 config](../../../config)
+to set the limits globally or refer to the
+[API of the Service builder](https://docs.rs/iceoryx2/latest/iceoryx2/service/index.html)
+to set them for a single service.
+
+## How to enable publish-subscribe communication between Rust, C++ and C applications
+
+To communicate with each other, publisher and subscriber applications must share
+the same service configuration, including the payload and the user header type
+name.
+
+For C applications, these type names must be set with
+`iox2_service_builder_pub_sub_set_payload_type_details` and
+`iox2_service_builder_pub_sub_set_user_header_type_details` before creating the
+service.
+
+For C++ and Rust applications, the internally derived type names
```

Comment:
Shouldn't this be only in the C++ example readme?

Reply:
Removed.